### ISSUE TITLE
in forward of wdsr.py, where dose 127.5 come from?

### ISSUE THREAD

**Issue by vegetable09** [JiahuiYu/wdsr_ntire2018#44]
In #40, you said you are doing data normaliztion. but i have no idea where 127.5 come from. 

as far as i know, data normalization is something like y = (x - μ)/σ, where μ is mean, and σ is the standard viarence. I guess [r, g, b]_means you used here may be the avg value for DIV2K_train_LR_bicubic:

R - 0.4488,
G - 0.4371,
B - 0.4040,

However, in forward, you used x = (x - self.rgb_mean.cuda()*255)/127.5
so i am confused where the σ = 127.5 come from? it should be [r, g, b]_σ but here is only one σ 

And another question is which dataset to use when calculate mean RGB, HR or LRx2, x3, x4. You said in the Experimental Results of your paper that you subtracted the mean RGB of DIV2K training set. I have tested that the mean value is different between HR and LRx2,x3,x4.

**Comment by vegetable09** [comment]
import torch
import time

import utility
import data
import model
import loss
from option import args
from trainer import Trainer

# 准备好loader
loader = data.Data(args)
r_mean = 0
g_mean = 0
b_mean = 0
pixels = 0
# 计算RGB图像的每个channel的均值和方差
loader = loader.loader_train
print(len(loader))
start = time.time()

# 调试时，number为0时，把_, 注释掉
for batch, (lr, hr, _, idx_scale) in enumerate(loader):
    print(batch)
    lr = lr.squeeze()
    pixels += lr.shape[1] * lr.shape[2]
    r, g, b = lr[0], lr[1], lr[2]
    r_mean += torch.sum(r)
    g_mean += torch.sum(g)
    b_mean += torch.sum(b)
print(time.time() - start)
r_mean /=  pixels
g_mean /=  pixels
b_mean /=  pixels
# 标准差
r_std = 0
g_std = 0
b_std = 0
pixels = 0
start = time.time()
for batch, (lr, hr, _, idx_scale) in enumerate(loader):
    print(batch)
    lr = lr.squeeze()
    pixels += lr.shape[1] * lr.shape[2]
    r, g, b = lr[0], lr[1], lr[2]
    r_std += torch.sum((r-r_mean)*(r-r_mean))
    g_std += torch.sum((g-g_mean)*(g-g_mean))
    b_std += torch.sum((b-b_mean)*(b-b_mean))
print(time.time() - start)
r_std = (r_std / pixels) ** 0.5
g_std = (g_std / pixels) ** 0.5
b_std = (b_std / pixels) ** 0.5
print(r_mean, g_mean, b_mean, r_std, g_std, b_std)

**Comment by vegetable09** [comment]
i use cmd `/workdir/EDSR-PyTorch/src/div2k_statistic.py --n_threads 0 --epochs 1 --batch_size 1 --dir_data /sr_data/datasets/super_resolution/ --data_train DIV2K --data_test "" --data_range 1-800/801-900 --scale 2` and result are following: 
lr r,g,b mean std: ['0564x2'] tensor(116.5574) tensor(110.7386) tensor(101.0412) tensor(70.3525) tensor(66.0828) tensor(71.0180)
hr r,g,b mean std: ['0564x2'] tensor(116.5522) tensor(110.7315) tensor(101.0332) tensor(71.3605) tensor(67.1300) tensor(71.9023)

**Comment by rongleiji** [comment]
Hi,  I think it is the total std of 0-255 , not of each channel.

**Comment by vegetable09** [comment]
> Hi, I think it is the total std of 0-255 , not of each channel.

got it, i will try and compare between using total std and per_channel std
